clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -329,6 +382,8 @@ void testEnsureEventsAreCompleted() {
 
     @Test
     void testCleanupInvokesReaper() {
+        LinkedList<NetworkClientDelegate.UnsentRequest> queue = new LinkedList<>();
+        when(networkClientDelegate.unsentRequests()).thenReturn(queue);
```

Comment:
totally needed, but heads-up, there is another [PR](https://github.com/apache/kafka/pull/16156) in-flight where we're trying to change the check behind this, so we'll need to update this expectation here if that goes in first.

Reply:
Thanks for the heads up, I will keep that in mind

Reply:
it just got merged, so worth rebasing and updating the test as needed ;)

Reply:
I will get that done, thank you for letting me know!